tests/test_user_mgt.py

```diff
@@ -105,14 +117,16 @@ def _check_user_record(user, expected_uid='testuser'):
     assert provider.provider_id == 'phone'
 
 
-def _check_request(recorder, want_url, want_body=None):
+def _check_request(recorder, want_url, want_body=None, want_timeout=None):
     assert len(recorder) == 1
     req = recorder[0]
     assert req.method == 'POST'
     assert req.url == '{0}{1}'.format(USER_MGT_URL_PREFIX, want_url)
     if want_body:
         body = json.loads(req.body.decode())
         assert body == want_body
+    if want_timeout:
+        assert recorder[0]._extra_kwargs['timeout'] == pytest.approx(want_timeout, 0.001)
```

Comment:
Should we add the following assertion?

```py
timeout_arg = recorder[0]._extra_kwargs['timeout']
if want_timeout:
    assert timeout_arg == pytest.approx(want_timeout, 0.01)
else:
    assert timeout_arg == pytest.approx(_http_client.DEFAULT_TIMEOUT, 0.01)
```

Reply:
The problem I see with that is it would force any test not using the default to pass that argument. I aligned with the other `want` args, if it's specified, check it, otherwise don't. That said, it's your code base so I defer to you. Lmk if you want me to change it.

Reply:
> The problem I see with that is it would force any test not using the default to pass that argument.

It doesn't. It can default to `None`, in which case we assert against `_http_client.DEFAULT_TIMEOUT`.

Reply:
That's what I'm saying, if someone added a test using `user_mgt_app_with_timeout`, but didn't pass a timeout to this function (presumably because they didn't care about asserting the value), it would fail.

Reply:
In the implementation we already default the client to `_http_client.DEFAULT_TIMEOUT`. In what scenario do we expect that assertion to not hold?

Reply:
If an app is used with a non-default timeout, but someone is either unaware of the `want_timeout` parameter or explicitly does not wish to make an assertion around it. Just biasing towards explicit > implicit [per BDFL](https://www.python.org/dev/peps/pep-0020/). Implementing the suggested behavior would implicitly check timeout. It also adds assertions to existing tests, whereas the original implementation results in no change to existing tests/assertions.

For a concrete example, if I removed the `TEST_TIMEOUT` from `test_get_user_with_timeout`, the test would fail, but, IMO, it would not be readily obvious just inspecting the test function why (nowhere did I make an assertion or ask an assertion to be made around timeout, so why is that happening?).

Reply:
> It also adds assertions to existing tests, whereas the original implementation results in no change to existing tests/assertions.

That was my intended goal. Ideally, the `_check_request()` helper function should make assertions about the traits that are common to all requests made by the auth client. We probably should have had the timeout assertion there to begin with. The `want_body` shouldn't be an optional argument. It's always specified.

Anyway, all these changes can probably be lumped into a separate PR. For now I'm happy with where this stands.